src/QsCompiler/Transformations/CallGraph/CallGraphBuilder.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+#nullable enable
+
+namespace Microsoft.Quantum.QsCompiler.DependencyAnalysis
+{
+    using Range = DataTypes.Range;
+
+    /// <summary>
+    /// Base class for call graph edge types.
+    /// </summary>
+    public abstract class CallGraphEdgeBase : IEquatable<CallGraphEdgeBase>
+    {
+        /// <summary>
+        /// Name of the callable where the reference was made.
+        /// </summary>
+        public QsQualifiedName FromCallableName { get; }
+
+        /// <summary>
+        /// Name of the callable being referenced.
+        /// </summary>
+        public QsQualifiedName ToCallableName { get; }
```

Comment:
We should maybe address this one in the PR that makes use of the VerifyAllCycles (#621) to see if we can remove the need to embed the FromCallableName and ToCallableName from the edge. Let's move this discussion there.

Reply:
I believe these properties were just added in this PR, is that correct? If so, I would prefer to address it in this PR. The idea that I had in mind would not require changing the signature of `VerifyAllCycles`. My idea was to change `GetEdges` to return `IEnumerable<IEnumerable<(SimpleCallGraphEdge, QsQualifiedName)>>` instead of simply `IEnumerable<IEnumerable<SimpleCallGraphEdge>>`.

Reply:
done